src/nvim/os/job.c

```diff
@@ -55,6 +56,22 @@ uv_signal_t schld;
 #endif
 // Callbacks for libuv
 
+static int count = 0;  // A cyclical counter used for assigning job IDs.
+
+/// Returns a new job ID within the range `[1, INT_MAX]`.
+///
+/// This number will be unique between all jobs that have not been stopped, but
+/// cycle back to one after reaching `INT_MAX`. This makes it highly unlikely
+/// for two jobs to have the same ID.
+static int new_job_id(void)
+{
+  count = count < INT_MAX ? count + 1 : 1;
```

Comment:
Is `0` a special value for `count`? Otherwise we could use an unsigned type for `count` and get the cyclical counter with just `count++`.